test/components/views/dialogs/AccessSecretStorageDialog-test.tsx

```diff
@@ -15,113 +15,97 @@ limitations under the License.
-        isValidRecoveryKey: jest.fn(),
-    });
+    let mockClient: MatrixClient;
+
     const defaultProps = {
         onFinished: jest.fn(),
         checkPrivateKey: jest.fn(),
         keyInfo: undefined,
     };
-    const getComponent = (props = {}): ReactWrapper =>
-        mount(<AccessSecretStorageDialog {...defaultProps} {...props} />);
 
-    beforeEach(() => {
-        jest.clearAllMocks();
-        mockClient.keyBackupKeyFromRecoveryKey.mockReturnValue("a raw key" as unknown as Uint8Array);
-        mockClient.isValidRecoveryKey.mockReturnValue(false);
+    const renderComponent = (props = {}): void => {
+        render(<AccessSecretStorageDialog {...defaultProps} {...props} />);
+    };
+
+    const enterSecurityKey = async (placeholder = "Security Key"): Promise<void> => {
+        await act(async () => {
+            fireEvent.change(screen.getByPlaceholderText(placeholder), {
+                target: {
+                    value: securityKey,
+                },
+            });
+            // wait for debounce
+            jest.advanceTimersByTime(250);
+        });
+    };
+
+    const submitDialog = async (): Promise<void> => {
+        await userEvent.click(screen.getByText("Continue"), { delay: null });
+    };
+
+    beforeAll(() => {
+        jest.useFakeTimers();
+        mockPlatformPeg();
     });
 
     afterAll(() => {
-        unmockClientPeg();
+        jest.useRealTimers();
+        jest.restoreAllMocks();
+    });
+
+    beforeEach(() => {
+        mockClient = getMockClientWithEventEmitter({
+            keyBackupKeyFromRecoveryKey: jest.fn(),
+            checkSecretStorageKey: jest.fn(),
+            isValidRecoveryKey: jest.fn(),
+        });
     });
 
     it("Closes the dialog when the form is submitted with a valid key", async () => {
+        mocked(mockClient.checkSecretStorageKey).mockResolvedValue(true);
+        mocked(mockClient.isValidRecoveryKey).mockReturnValue(true);
+
         const onFinished = jest.fn();
         const checkPrivateKey = jest.fn().mockResolvedValue(true);
-        const wrapper = getComponent({ onFinished, checkPrivateKey });
-
-        // force into valid state
-        act(() => {
-            wrapper.setState({
-                recoveryKeyValid: true,
-                recoveryKey: "a",
-            });
-        });
-        const e = { preventDefault: () => {} };
+        renderComponent({ onFinished, checkPrivateKey });
 
-        act(() => {
-            wrapper.find("form").simulate("submit", e);
-        });
+        // check that the input field is focused
+        expect(screen.getByPlaceholderText("Security Key")).toHaveFocus();
 
-        await flushPromises();
+        await enterSecurityKey();
+        await submitDialog();
 
-        expect(checkPrivateKey).toHaveBeenCalledWith({ recoveryKey: "a" });
-        expect(onFinished).toHaveBeenCalledWith({ recoveryKey: "a" });
+        expect(screen.getByText("Looks good!")).toBeInTheDocument();
```

Comment:
```suggestion
        screen.getByText("Looks good!");
```

By definition anything returned by `screen.getByText` is on screen and in the document

Reply:
Yes, but in my opinion this is less explicitly. Did someone forget to trigger an action on it? Where `toBeInTheDocument` clearly says - jep should be there.

I would personally go for `toBeInTheDocument` unless this has a performance impact.

Reply:
OK, not a big deal either way